sdk/python/kfp/v2/compiler/compiler_utils.py

```diff
@@ -82,13 +82,8 @@ def validate_pipeline_name(name: str) -> None:
 
 def is_v2_component(op: _container_op.ContainerOp) -> bool:
   """Determines whether a component is a KFP v2 component."""
-  if not op._metadata or not op._metadata.metadata:
-    return False
-  if not (op._metadata.metadata.annotations or _component_builder
-          .V2_COMPONENT_ANNOTATION not in op._metadata.metadata.annotations):
-    return False
-  return bool(op._metadata.metadata.annotations[
-      _component_builder.V2_COMPONENT_ANNOTATION])
+  component_spec = op._metadata
+  return component_spec and component_spec.metadata and component_spec.metadata.annotations.get(_component_builder.V2_COMPONENT_ANNOTATION) == 'true'
```

Comment:
nit:  can you break this line to fit under 80 characters?

Reply:
Done.